test/benchmarks/ml/vllm_test.go

```diff
@@ -0,0 +1,174 @@
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types/mount"
+	"gvisor.dev/gvisor/pkg/test/dockerutil"
+	"gvisor.dev/gvisor/test/benchmarks/harness"
+)
+
+func TestMain(m *testing.M) {
+	harness.Init()
+	harness.SetFixedBenchmarks()
+	os.Exit(m.Run())
+}
+
+// BenchmarkVLLM runs a vLLM workload.
+func BenchmarkVLLM(b *testing.B) {
+	doVLLMTest(b)
+}
+
+func doVLLMTest(b *testing.B) {
+	serverMachine, err := harness.GetMachine()
+	if err != nil {
+		b.Fatalf("failed to get machine: %v", err)
+	}
+	defer serverMachine.CleanUp()
+
+	b.Run("opt-125", func(b *testing.B) {
+		ctx := context.Background()
+
+		b.ResetTimer()
+		b.StopTimer()
+
+		for i := 0; i < b.N; i++ {
+			serverCtr := serverMachine.GetNativeContainer(ctx, b)
+			defer serverCtr.CleanUp(ctx)
+			if err := harness.DropCaches(serverMachine); err != nil {
+				b.Skipf("failed to drop caches: %v. You probably need root.", err)
+			}
+
+			// Run vllm.
+			runOpts := dockerutil.GPURunOpts()
+			runOpts.Image = "benchmarks/vllm"
+			runOpts.Env = []string{"PYTHONPATH=$PYTHONPATH:/vllm"}
+
+			if err := serverCtr.Spawn(ctx, runOpts); err != nil {
+				b.Errorf("failed to run container: %v", err)
+			}
+			if out, err := serverCtr.WaitForOutput(ctx, "Uvicorn running on http://0.0.0.0:8000", 10*time.Minute); err != nil {
+				b.Fatalf("failed to start vllm model: %v %s", err, out)
+			}
+
+			clientMachine, err := harness.GetMachine()
+			if err != nil {
+				b.Fatalf("failed to get machine: %v", err)
+			}
+			defer clientMachine.CleanUp()
+			clientCtr := clientMachine.GetNativeContainer(ctx, b)
+			defer clientCtr.CleanUp(ctx)
+
+			b.StartTimer()
+
+			// store vllm logs here
+			logsDir := b.TempDir()
+
+			out, err := clientCtr.Run(ctx, dockerutil.RunOpts{
+				Links: []string{serverCtr.MakeLink("vllmctr")},
+				Image: "benchmarks/vllm",
+				Env:   []string{"PYTHONPATH=$PYTHONPATH:/vllm"},
+				Mounts: []mount.Mount{
+					// The logs dir is used because vllm only outputs json to a file.
+					{
+						Source: logsDir,
+						Target: "/tmp",
+						Type:   "bind",
+					},
+				},
+			}, "/vllm/benchmarks/benchmark_serving.py", "--host", "vllmctr", "--model", "/model", "--tokenizer", "/model", "--endpoint", "/v1/completions", "--backend", "openai", "--dataset", "/ShareGPT_V3_unfiltered_cleaned_split.json", "--save-result", "--result-dir", "/tmp")
+			if err != nil {
+				b.Errorf("failed to run container: %v logs: %s", err, out)
+			}
+
+			b.StopTimer()
+
+			metrics, err := parseVLLMJSON(logsDir)
+			if err != nil {
+				b.Errorf("failed to parse vllm output: %v", err)
+			}
+			b.ReportMetric(float64(metrics.Completed), "requests")
+			b.ReportMetric(metrics.RequestThroughput, "request_throughput")
+			b.ReportMetric(metrics.InputThroughput, "input_tok_throughput")
+			b.ReportMetric(metrics.OutputThroughput, "output_tok_throughput")
+			b.ReportMetric(metrics.MedianTTFTMS, "median_ttft_ms")
```

Comment:
For consistency with how other gVisor benchmark metrics are reported (though I'll grant that they're not all consistent...), please use a format like `${metric_name}-${statistic}-${unit}`. For example here this would be `ttft-p50-ns`.